stubs/setuptools/pkg_resources/__init__.pyi

```diff
@@ -18,8 +17,10 @@ _InstallerType: TypeAlias = Callable[[Requirement], Distribution | None]
 _EPDistType: TypeAlias = Distribution | Requirement | str
 _MetadataType: TypeAlias = IResourceProvider | None
 _PkgReqType: TypeAlias = str | Requirement
-_DistFinderType: TypeAlias = Callable[[_Importer, str, bool], Generator[Distribution, None, None]]
-_NSHandlerType: TypeAlias = Callable[[_Importer, str, str, types.ModuleType], str]
+_ModuleLike: TypeAlias = object | types.ModuleType  # Any object that optionally has __loader__ or __file__, usually a module
+_ProviderFactoryType: TypeAlias = Callable[[_ModuleLike], IResourceProvider]
+_DistFinderType: TypeAlias = Callable[[_T, str, bool], Iterable[Distribution]]
```

Comment:
If it wasn't for the `find_nothing` method that returns an empty tuple, this could correctly return a Generator.